### PR TITLE
Workbench default layout

### LINKED ISSUE #21566: Workbench: Default layout of initial dock widgets
Define the default layout for the workbench widets on first run. It should bring together the current work on the workspaces widget, algorithms tool box, IPython shell, scripting environment and log message window.

The layout should look roughly something like:

main-layout-workspaces-left

The widget on the left is supposed to be a new workspace tree widget but the old one will have to suffice for now.

Also, create a Window menu that will allow users to show/hide the docks along with a "Restore Default Layout" action to reset to "factory" settings.

### PR DESCRIPTION
**Pull request by ElliotAOram**
* Added the weighting fraction for height and width of columns/rows in the workbench.
* Set up the default layout to be correct upon launch
* Ensured that the max width/height is taken from full screen area upon launch (previously relied on defaults in `CONF`)
* Added a `View > Restore Default Layout` button in the QMenu which reverts the current state of the interface to what you see on launch.  

**To test:**

* Run the workbench and check the layout is as expected (described in the issue)
* Check you can drag the vertical and horizontal boundaries of each widget and all widget movement is as expected.
* Delete/Move/Reorder widget and the press the `View > Restore Default Layout` button in the menu. (Do the same but with `Shift+F10`: shortcut for the button)   

Fixes #21566

**Release Notes** 
*Does not need to be in the release notes.*

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####

- [ ] Is the code of an acceptable quality?
- [ ] Does the code conform to the [coding standards](http://www.mantidproject.org/Coding_Standards)?
- [ ] Are the unit tests small and test the class in isolation?
- [ ] If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- [ ] Do changes function as described? Add comments below that describe the tests performed?
- [ ] Do the changes handle unexpected situations, e.g. bad input?
- [ ] Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

**Review by martyngigg — CHANGES_REQUESTED** [review]
The layout looks as requested :+1: 

A few small things:

* There seems to be an issue with resizing column 0 - if I try to drag it smaller it gets stuck at it's current width. Also if you close the algorithm selector then the workspace widget doesn't resize vertically to fill the space.
* The View menu needs the list of possible docks so that users can toggle them on and off easily.

**Comment by martyngigg** [comment]
This is looking better. The view menu items work as expected and the other widgets resize when a dock is closed as expected.
One final issue seems to be that when the algorithm selector is visible you can't resize column 0 smaller some fixed amount. Could we look into that? Maybe there is a minimum width hardcoded somewhere like there was with the workspace widget?

**Comment by ElliotAOram** [comment]
Issue for resolving temp fix implemented here #21611

**Comment by ElliotAOram** [comment]
@martyngigg I wasn't able to reproduce your issue with the incorrect scaling on start up (tested on both my windows machines). I'll have a quick check on my Ubuntu VM to see if it's an OS thing

**Comment by ElliotAOram** [comment]
No luck on Ubuntu either. @samueljackson92 has kindly said he'll take a look on OSX

File: qt/applications/workbench/workbench/app/mainwindow.py [hunk 1]
```diff
@@ -352,19 +319,58 @@ def setup_default_layouts(self, window_settings):
...
-        CONF.set(section, prefix + 'size', (win_size.width(), win_size.height()))
-        CONF.set(section, prefix + 'is_maximized', self.isMaximized())
-        CONF.set(section, prefix + 'is_fullscreen', self.isFullScreen())
-        pos = self.window_position
-        CONF.set(section, prefix + 'position', (pos.x(), pos.y()))
-        self.maximize_dockwidget(restore=True)  # Restore non-maximized layout
-        qba = self.saveState()
-        CONF.set(section, prefix + 'state', qbytearray_to_str(qba))
+            # set the width and height
+            self._layout_widget_info = []
```
Comment (martyngigg) [inline qt/applications/workbench/workbench/app/mainwindow.py]: As discussed offline. This doesn't work properly for X11 (for some reason) so I think it's best to revisit it after the initial demo as without the layout is nearly what we want anyway.

**Review by martyngigg — APPROVED** [review]

**Pull request merged**